src/modules/Sticky/Sticky.js

```diff
@@ -0,0 +1,245 @@
+     * @param {object} data - All props.
+     */
+    onBottom: PropTypes.func,
+
+    /**
+     * Callback when element is fixed to page.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     */
+    onStick: PropTypes.func,
+
+    /**
+     * Callback when element is bound to top of parent container.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     */
+    onTop: PropTypes.func,
+
+    /**
+     * Callback when element is unfixed from page.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     */
+    onUnstick: PropTypes.func,
+
+    /** Whether element should be "pushed" by the viewport, attaching to the bottom of the screen when scrolling up. */
+    pushing: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    bottomOffset: 0,
+    offset: 0,
+  }
+
+  static _meta = {
+    name: 'Sticky',
+    type: META.TYPES.MODULE,
+  }
+
+  state = {
+    sticky: false,
+  }
+
+  componentDidMount() {
+    this.handleUpdate()
+    window.addEventListener('scroll', this.handleUpdate)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.handleUpdate)
+  }
+
+  // ----------------------------------------
+  // Handlers
+  // ----------------------------------------
+
+  handleUpdate = (e) => {
+    const { pushing } = this.state
+
+    this.assignRects()
+
+    if (pushing) {
+      if (this.reachedStartingPoint()) return this.stickToContextTop(e)
+      if (this.touchedScreenBottom()) return this.stickToScreenBottom(e)
+      return this.stickToContextBottom(e)
+    }
+
+    if (this.oversized()) {
+      if (this.contextRect.top > 0) return this.stickToContextTop(e)
+      if (this.contextRect.bottom < window.innerHeight) return this.stickToContextBottom(e)
+    }
+
+    if (this.touchedScreenTop()) {
+      if (this.reachedContextBottom()) return this.stickToContextBottom(e)
+      return this.stickToScreenTop(e)
+    }
+
+    return this.stickToContextTop(e)
+  }
+
+  // ----------------------------------------
+  // Helpers
+  // ----------------------------------------
+
+  assignRects = () => {
+    const { context } = this.props
+
+    this.triggerRect = this.triggerRef.getBoundingClientRect()
+    this.contextRect = (context || document.body).getBoundingClientRect()
+    this.stickyRect = this.stickyRef.getBoundingClientRect()
+  }
+
+  computeStyle() {
+    const { bottom, sticky, top } = this.state
+
+    if (!sticky) return {}
+    return {
+      bottom,
+      top,
+      position: 'fixed',
+      width: this.triggerRect.width,
+    }
+  }
+
+  // Return true if the height of the component is higher than the window
+  oversized = () => this.stickyRect.height > window.innerHeight
+
+  // Return true when the component reached the bottom of the context
+  reachedContextBottom = () => {
```

Comment:
I removed the auxiliary verbs from the names after @layershifter's code review 😅 . Shall I put them back there?